packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -307,38 +308,38 @@ class DebuggerController extends DisposableController
   ValueListenable<List<StackFrameAndSourcePosition>>
       get stackFramesWithLocation => _stackFramesWithLocation;
 
-  final _selectedStackFrame = ValueNotifier<StackFrameAndSourcePosition>(null);
+  final _selectedStackFrame = ValueNotifier<StackFrameAndSourcePosition?>(null);
 
-  ValueListenable<StackFrameAndSourcePosition> get selectedStackFrame =>
+  ValueListenable<StackFrameAndSourcePosition?> get selectedStackFrame =>
       _selectedStackFrame;
 
-  Frame get frameForEval =>
+  Frame? get frameForEval =>
       _selectedStackFrame.value?.frame ??
-      _stackFramesWithLocation.value?.safeFirst?.frame;
+      _stackFramesWithLocation.value.safeFirst?.frame;
 
   final _variables = ValueNotifier<List<DartObjectNode>>([]);
 
   ValueListenable<List<DartObjectNode>> get variables => _variables;
 
-  final _breakpoints = ValueNotifier<List<Breakpoint>>([]);
+  final _breakpoints = ValueNotifier<List<Breakpoint>?>([]);
 
-  ValueListenable<List<Breakpoint>> get breakpoints => _breakpoints;
+  ValueListenable<List<Breakpoint?>?> get breakpoints => _breakpoints;
 
   final _breakpointsWithLocation =
       ValueNotifier<List<BreakpointAndSourcePosition>>([]);
 
   ValueListenable<List<BreakpointAndSourcePosition>>
       get breakpointsWithLocation => _breakpointsWithLocation;
 
-  final _selectedBreakpoint = ValueNotifier<BreakpointAndSourcePosition>(null);
+  final _selectedBreakpoint = ValueNotifier<BreakpointAndSourcePosition?>(null);
 
-  ValueListenable<BreakpointAndSourcePosition> get selectedBreakpoint =>
+  ValueListenable<BreakpointAndSourcePosition?> get selectedBreakpoint =>
       _selectedBreakpoint;
 
   final _exceptionPauseMode =
-      ValueNotifier<String>(ExceptionPauseMode.kUnhandled);
+      ValueNotifier<String?>(ExceptionPauseMode.kUnhandled);
 
-  ValueListenable<String> get exceptionPauseMode => _exceptionPauseMode;
+  ValueListenable<String?> get exceptionPauseMode => _exceptionPauseMode;
```

Comment:
we should be able to make these String types non-nullable by defaulting to `ExceptionPauseMode.kUnhandled` everywhere where a null value may have been assigned.